packages/validator-ajv8/src/validator.ts

```diff
@@ -0,0 +1,383 @@
+import Ajv, { ErrorObject } from "ajv";
+import toPath from "lodash/toPath";
+import {
+  CustomValidator,
+  ErrorSchema,
+  ErrorTransformer,
+  FieldValidation,
+  FormValidation,
+  GenericObjectType,
+  RJSFSchema,
+  RJSFValidationError,
+  ValidationData,
+  ValidatorType,
+  getDefaultFormState,
+  isObject,
+  mergeValidationData,
+  ERRORS_KEY,
+  REF_KEY,
+} from "@rjsf/utils";
+
+import { CustomValidatorOptionsType } from "./types";
+import createAjvInstance from "./createAjvInstance";
+
+const ROOT_SCHEMA_PREFIX = "__rjsf_rootSchema";
+
+/** `ValidatorType` implementation that uses the AJV 6 validation mechanism.
+ */
+export default class AJV6Validator<T = any> implements ValidatorType<T> {
```

Comment:
You could try to have AJV6Validator and AJV8Validator inherit a shared prototype, but that might be messy

Reply:
It is very close to being identical, BUT it has a few differences needed by the migration. Moreover, once we start adding additional capabilities (compiling the schema, localization of error message, etc) this will veer more and more away from the AJV6 validator